Tests/Integration/LoRaDataRequestHandlerConnectionAffinityIntegrationTests.cs

```diff
@@ -64,55 +64,89 @@ public LoRaDataRequestHandlerConnectionAffinityIntegrationTests(ITestOutputHelpe
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        public async Task Connection_Handling_Should_Depend_On_Deduplication_Result_And_Processing_Delay(bool canProcess, bool processingDelayEnabled)
         {
             // arrange
             this.deviceMock.Object.IsConnectionOwner = true;
             _ = this.dataRequestHandlerMock.Setup(x => x.TryUseBundlerAssert()).Returns(new FunctionBundlerResult
             {
                 DeduplicationResult = new DeduplicationResult
-                { CanProcess = keptConnection },
+                { CanProcess = canProcess },
                 NextFCntDown = null
             });
 
+            if (!processingDelayEnabled)
+                ServerConfiguration.ProcessingDelayInMilliseconds = 0;
+
             // act
             _ = await this.dataRequestHandlerMock.Object.ProcessRequestAsync(this.loraRequest, this.deviceMock.Object);
 
             // assert
-            Assert.Equal(keptConnection, this.deviceMock.Object.IsConnectionOwner);
+            Assert.Equal(canProcess || !processingDelayEnabled, this.deviceMock.Object.IsConnectionOwner);
 
-            if (keptConnection)
+            if (canProcess)
             {
                 this.deviceMock.Verify(x => x.CloseConnectionAsync(CancellationToken.None, It.IsAny<bool>()), Times.Never);
                 this.deviceMock.Verify(x => x.BeginDeviceClientConnectionActivity(), Times.Once);
                 this.dataRequestHandlerMock.Verify(x => x.SaveChangesToDeviceAsyncAssert(), Times.Once);
             }
             else
             {
-                this.deviceMock.Verify(x => x.CloseConnectionAsync(CancellationToken.None, false), Times.Once);
+                if (processingDelayEnabled)
+                {
+                    this.deviceMock.Verify(x => x.CloseConnectionAsync(CancellationToken.None, false), Times.Once);
+                    this.dataRequestHandlerMock.Verify(x => x.SaveChangesToDeviceAsyncAssert(), Times.Never);
+                }
+                else
+                {
+                    this.deviceMock.Verify(x => x.CloseConnectionAsync(CancellationToken.None, It.IsAny<bool>()), Times.Never);
+                    this.dataRequestHandlerMock.Verify(x => x.SaveChangesToDeviceAsyncAssert(), Times.Once);
+                }
+
                 this.deviceMock.Verify(x => x.BeginDeviceClientConnectionActivity(), Times.Never);
-                this.dataRequestHandlerMock.Verify(x => x.SaveChangesToDeviceAsyncAssert(), Times.Never);
             }
         }
 
+        [Theory]
+        [InlineData(null, true)]
+        [InlineData(400, true)]
+        [InlineData(1500, true)]
+        [InlineData(0, false)]
+        [InlineData(-1000, false)]
+        public void IsProcessingDelayEnbled(int? processingDelay, bool processingDelayEnabled)
```

Comment:
[nit] This is more of a unit test I would say.

Reply:
That's what I thought as well but the test is using `TestDefaultLoRaRequestHandler` which is part of the integration tests project and requires setting up a number of dependencies before it can be used in the actual test code so I would argue it belongs in the integration tests.

Reply:
Done in 65f0f573b3322e0c3d80db70f4cd35a317973521, thanks for the good suggestion:)